### PR TITLE
move python-xml to raw_install_python.yml

### PR DESCRIPTION
**Pull request by johanneskastl**
The package python-xml is needed for ansible's zypper module to interact with
the zypper package management tool.

roles/ceph-defaults/defaults/main.yml:
Remove python-xml from variable suse_package_dependencies to only
install python-xml on SUSE/openSUSE if python is not found.
raw_install_python.yml already contains all the logic needed to check
if there is a valid python installation, so this is better suited there.

openSUSE Leap 15.x / SLES 15.x do no longer have /usr/bin/python,
only /usr/bin/python3, which already contains the xml module, so
nothing needs to be installed in that case.

### CODE REVIEW

File: raw_install_python.yml [hunk 1]
```diff
@@ -50,8 +50,25 @@
...
-      raw: zypper -n install python-base
+      raw: zypper -n install python-base python-xml
       register: result
       until: result is succeeded
       when: stat_zypper.rc == 0
-  when: not True in (systempython.results | selectattr('stat', 'defined') | map(attribute='stat.exists') | list | unique)
\ No newline at end of file
+  when: not True in (systempython.results | selectattr('stat', 'defined') | map(attribute='stat.exists') | list | unique)
+
+- name: check for python2 on SUSE/openSUSE
```
Comment (guits) [inline raw_install_python.yml]: this task is a duplicate of the first one

File: raw_install_python.yml [hunk 2]
```diff
@@ -50,8 +50,25 @@
...
+  stat:
+    path: /usr/bin/python
```
Comment (guits) [inline raw_install_python.yml]: how do you know whether this is python2 or python3 ?

**Comment by guits** [comment]
i'm -1 on merging this,
IMO, this is definitely not the best approach.
I think you should stick to handle this `python-xml` installation in ceph-common since you can use a condition like `ansible_python_version is version( '3', '<')`

File: raw_install_python.yml (hunk 2 above)
Comment (johanneskastl) [reply raw_install_python.yml]: On openSUSE Leap 15.x / SLES 15.x, only python2 provides `/usr/bin/python`.

File: raw_install_python.yml (hunk 1 above)
Comment (johanneskastl) [reply raw_install_python.yml]: Yes, but I was not sure if I could get only the detection of `/usr/bin/python` out of the first task.

**Comment by johanneskastl** [comment]
> i'm -1 on merging this,
> IMO, this is definitely not the best approach.
> I think you should stick to handle this `python-xml` installation in ceph-common since you can use a condition like `ansible_python_version is version( '3', '<')`

AFAIK python-xml is needed to install packages via the `zypper` module, that ansible uses as soon as you use a `package` task. So this needs to be installed using the raw module before anything else uses the `package` module.

Of course we could somehow add a raw install task later one, but IMHO this is the right place for raw tasks.

**Comment by guits** [comment]
> AFAIK python-xml is needed to install packages via the `zypper` module, that ansible uses as soon as you use a `package` task. 

ah yes, in fact, you already told this in the other PR, I forgot.
ok, in that case, yes, it's kind of egg or chicken issue, indeed.

well, in that case, you could at least reuse the first task.

File: raw_install_python.yml [hunk 3]
```diff
@@ -50,8 +50,25 @@
...
+      raw: zypper -n install python-base python-xml
       register: result
       until: result is succeeded
       when: stat_zypper.rc == 0
-  when: not True in (systempython.results | selectattr('stat', 'defined') | map(attribute='stat.exists') | list | unique)
\ No newline at end of file
+  when: not True in (systempython.results | selectattr('stat', 'defined') | map(attribute='stat.exists') | list | unique)
+
+- name: check for python2 on SUSE/openSUSE
+  stat:
```
Comment (guits) [inline raw_install_python.yml]:
```suggestion
```

File: raw_install_python.yml [hunk 4]
```diff
@@ -50,8 +50,25 @@
...
+  failed_when: false
+  register: systempython2
+  when: 
+    - stat_zypper.rc == 0
+
+- name: install python-xml for opensuse only if python2 is installed already
+  raw: zypper -n install python-xml
+  register: result
+  until: result is succeeded
+  when:
```
Comment (guits) [inline raw_install_python.yml]:
```suggestion
  with_items: "{{ systempython.results }}"
  when: 
```

File: raw_install_python.yml [hunk 5]
```diff
@@ -50,8 +50,25 @@
...
+  when: 
+    - stat_zypper.rc == 0
+
+- name: install python-xml for opensuse only if python2 is installed already
+  raw: zypper -n install python-xml
+  register: result
+  until: result is succeeded
+  when: 
+    - stat_zypper.rc == 0
+    - systempython2.stat.exists
```
Comment (guits) [inline raw_install_python.yml]:
```suggestion
    - item.stat.exists | bool
    - item.stat.path == '/usr/bin/python'
```

File: raw_install_python.yml (hunk 2 above)
Comment (guits) [inline raw_install_python.yml]: remove this task

File: raw_install_python.yml (hunk 3 above)
Comment (johanneskastl) [reply raw_install_python.yml]: Skipped, as I deleted the tasks as requested below

File: raw_install_python.yml (hunk 5 above)
Comment (johanneskastl) [reply raw_install_python.yml]: Done.

File: raw_install_python.yml (hunk 4 above)
Comment (johanneskastl) [reply raw_install_python.yml]: Done

File: raw_install_python.yml (hunk 2 above)
Comment (johanneskastl) [reply raw_install_python.yml]: Done

**Comment by johanneskastl** [comment]
Hi Guillaume, sorry it took so long to complete this. I followed your suggestions, please review again.

File: raw_install_python.yml (hunk 2 above)
Comment (guits) [reply raw_install_python.yml]: you can remove this task now

**Comment by johanneskastl** [comment]
> almost there, just run ./generate_group_vars_sample.sh

Done.

File: raw_install_python.yml [hunk 6]
```diff
@@ -56,3 +56,13 @@
...
       when: stat_zypper.rc == 0
   when: not True in (systempython.results | selectattr('stat', 'defined') | map(attribute='stat.exists') | list | unique)
+
+- name: install python-xml for opensuse only if python2 is installed already
+  raw: zypper -n install python-xml
+  register: result
+  until: result is succeeded
+  with_items: "{{ systempython.results }}"
+  when:
+    - stat_zypper.rc == 0
```
Comment (guits) [inline raw_install_python.yml]:
```suggestion
    - stat_zypper.rc is defined
    - stat_zypper.rc == 0
```

Comment (johanneskastl) [reply raw_install_python.yml]: Fixed.

**Pull request merged**